R/Tidy_names_types_factors.R

```diff
@@ -307,3 +324,43 @@ parse_date_columns <- function(data, date_formats) {
 
   return(vct)
 }
+
+
+#' Locate NA values introduced during data cleaning
+#'
+#' Finds and locates NA values that were introduced by calling `apply_data_dictionary()`
+#'   on a dataframe using a data_dictionary.
+#'
+#'
+#' @param data_raw data.frame that was provided as input to `apply_data_dictionary()`
+#' @param data data.frame that is returned by `apply_data_dictionary()`
+#' @param data_dictionary the data_dictionary used by `apply_data_dictionary()`
+#'   to turn "data_raw" into "data".
+#'
+#' @return returns a dataframe with the location of introduced NA's. If no NA's
+#'   were introduced an empty dataframe is returned.
+#'
+#' @keywords internal
+#'
+#' @author Till D. Best, J. Peter Marquardt
+.find_NA_coercions <- function(data_raw, data, data_dictionary) {
+  #  create a dataframe matchin the old and new column name in our data dictionary
+  rosetta_stone <- data.frame(old_name = data_dictionary$old_column_name[!is.na(data_dictionary$new_data_type)],
+                              new_name = colnames(data))
+
+  # find those value where NA's were introduced
+  NA_difference <- is.na(data_raw[rosetta_stone$old_name]) < is.na(data)
```

Comment:
If we made this a parameter, could we use this method in a more general application? That might actually be worth considering

Reply:
see my general comment